subsys/suit/mci/Kconfig

```diff
@@ -20,6 +20,11 @@ config SUIT_MCI_IMPL_NRF54H20_SDFW
 	bool "nRF54H20: Secure domain"
 	depends on SUIT_PLATFORM_VARIANT_SDFW
```

Comment:
needs nrf54h20 guard

Reply:
I'd not agree here. We have cases, where we use the 54 implementation on POSIX to run tests.

Reply:
`depends on <nrf54h20> || <posix>`?

Reply:
And actually tests should not need this because they should force config e.g. https://github.com/nrfconnect/sdk-nrf/blob/main/tests/lib/modem_info/CMakeLists.txt there is a similar thing for network tests